trinity/tx_pool/validators.py

```diff
@@ -0,0 +1,59 @@
+from typing import (
+    Type
+)
+
+from evm.chains.base import (
+    BaseChain
+)
+from evm.exceptions import (
+    ValidationError
+)
+from evm.rlp.transactions import (
+    BaseTransaction,
+    BaseTransactionFields
+)
+from evm.vm.base import (
+    BaseVM
+)
+
+
+class DefaultTransactionValidator():
+    """
+    The :class:`~trinity.tx_pool.validators.DefaultTransactionValidator` class is responsible to
+    decide wether transactions should be relayed to peers or not. This implementation validates
+    transactions against an initially provided transaction class but will switch to a different
+    one as soon as the tip of the chain uses a more up to date transaction class than the one
+    that was initially provided.
+    """
+
+    def __init__(self, chain: BaseChain, initial_tx_class: Type[BaseTransaction]) -> None:
+        self.chain = chain
+        self._initial_tx_class = initial_tx_class
```

Comment:
If we instead used a `block_number` to specify which vm should be used, then we could eliminate the need to validate whether the `initial_tx_class` is present in the `vm_configuration` and just extract the appropriate *initial* vm from the `vm_configuration` based on the provided block number.

Reply:
I like that :+1: